.eslintrc.js

```diff
@@ -117,6 +117,10 @@ module.exports = {
 				selector: 'CallExpression[callee.name=/^(invokeMap|get|has|hasIn|invoke|result|set|setWith|unset|update|updateWith)$/] > Literal:nth-child(2)',
 				message: 'Always pass an array as the path argument',
 			},
+			{
+				selector: 'CallExpression[callee.name=/^(property|matchesProperty|path)$/] > Literal:nth-child(1)',
```

Comment:
Minor: An alternative would be to change the previous selector to include this cases e.g:
```
{
	selector: 'CallExpression[callee.name=/^(invokeMap|get|has|hasIn|invoke|result|set|setWith|unset|update|updateWith)$/] > Literal:nth-child(2), ' +
		'CallExpression[callee.name=/^(property|matchesProperty|path)$/] > Literal:nth-child(1)',
	message: 'Always pass an array as the path argument',
},
```

The main advantage would be that the message is not duplicated, but the selector looks complex and I'm not totally sure if this version is better or not.

Reply:
Hmm, it's a good point. Since we're in JS here, I wonder if we could generate this to a single rule dynamically with a `map`:

```js
[
	[ 1, [
		'property',
		'matchesProperty',
		'path',
	] ],
	[ 2, [
		'invokeMap',
		'get',
		'has',
		'hasIn',
		'invoke',
		'result',
		'set',
		'setWith',
		'unset',
		'update',
		'updateWith',
	] ],
].map( ( selector, [ argPosition, fnNames ] ) => (
	`CallExpression[callee.name=/^(${ fnNames.join( '|' ) })$/] > Literal:nth-child(${ argPosition })`
) ).join( ',' )
```

Or maybe too [clever](https://softwareengineering.stackexchange.com/questions/25276/why-is-cleverness-considered-harmful-in-programming-by-some-people)? 😆 Just trying to avoid the excessive duplication for what's effectively only varying on argument position and the function names.

Reply:
Hi, I personally think this code not too complex and we are not being too clever. Also, my expectation is that will not have to deal with this rules so I think this version is totally acceptable.
The only "improvement" I'm seeing would be to use objects instead of arrays, something like:
```
[ {
argPosition: 1,
functionNames: [ 'property','matchesProperty', 'path' ]
},
```
...

Maybe an object makes things even simpler.

Reply:
In hindsight, an object would be much easier / clearer than working with the tuples, aside from minor detail of needing to pull in Lodash's `map`. Pushed updated in c9ab732.